Hardcore.lua

```diff
@@ -250,7 +251,7 @@ function Hardcore:PLAYER_LOGIN()
 			time_played = 0,
 			accumulated_time_diff = 0,
 			tracked_played_percentage = 0,
-			deaths = 0,
+			deaths = {},
```

Comment:
I like this solution! I am concerned about future-proofing edits to Hardcore_Character types.  Does this cause an error down the line if we are trying to index based on player_alive_trigger etc..

Reply:
Not sure what you mean, do you have an example of what you think could go wrong?

Reply:
Yeah, its not really an issue at the moment or with this PR but if we have version releases where Hardcore_Character.var changes from an int to a dictionary, that var will remain as an int since its already initialized. Later code paths might try to use that var as a dict, Hardcore_Character.var.key.

Reply:
Ok I understand what you mean now!
If we want to change the data type of saved variables, we have to add a legacy case to handle the old data type.
For example in this case, we could check if Harcore_Character.deaths is of type number, and if it is we create a table with n records where n is the number saved in Hardcore_Character.deaths, and only run code that requires deaths to be a table after.

Might actually be worth adding it even for this, even though it is not an issue on release, but the streamers who have been testing the addon will get an error if they log onto characters that have deaths saved as a number.

Reply:
It's pretty straight-forward to put data-type cleaning as part of our overall library architecture - read cleanly as both old and new types, but always write new type

Reply:
Seems like potentially a lot of cases to cover. What if we used a more modular structure in the future so we don't have to add bridges for releases? I think the dict struct that rei is using is pretty future proof

Reply:
any time we change a typed variable, we'd have to use a method, e.g. HCDB:SetValue('whatever', somevalue)
any time we want to use it, we'd use something different, e.g. HCDB:GetValue('whatever', optional_default)

the problem solves itself over time